### PR TITLE
Add DelegateCommand<T>

### PR DESCRIPTION
**Pull request by baptiste-nv**
GitHub Issue (If applicable): #
None

## PR Type
What kind of change does this PR introduce?
Feature

## What is the current behavior?
No DelegateCommand that take a parameter

## What is the new behavior?
Add DelegateCommand<T>

## PR Checklist
- [X] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Contains **NO** breaking changes
- [X] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/Uno.Foundation/DelegateCommand.T.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		}
+
+		public bool CanExecute(object parameter)
+		{
+			return true;
+		}
+
+		public void Execute(object parameter)
+		{
+			_action?.Invoke((T)parameter);
```
Comment (jeromelaban) [inline src/Uno.Foundation/DelegateCommand.T.cs]: Make it visible that parameter may not be of the proper type

File: src/Uno.Foundation/DelegateCommand.T.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	public class DelegateCommand<T> : ICommand
+	{
+		private Action<T> _action;
+
+		public event EventHandler CanExecuteChanged;
+
+		public DelegateCommand(Action<T> action)
+		{
+			_action = action;
+			OnCanExecuteChanged(true);
```
Comment (jeromelaban) [inline src/Uno.Foundation/DelegateCommand.T.cs]: CanExecuteChanged cannot have been registered at this point.

File: src/Uno.Foundation/DelegateCommand.T.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Windows.Input;
+
+namespace Uno.UI.Common
+{
+	public class DelegateCommand<T> : ICommand
```
Comment (jeromelaban) [inline src/Uno.Foundation/DelegateCommand.T.cs]: There's some missing green, and unit tests.

File: src/Uno.Foundation/DelegateCommand.T.cs (hunk 2 above)
Comment (baptiste-nv) [reply src/Uno.Foundation/DelegateCommand.T.cs]: Copy/paste from DelegateCommand, I'll fix both

File: src/Uno.Foundation/DelegateCommand.T.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		}
+
+		public bool CanExecute(object parameter)
+		{
+			return true;
+		}
+
+		public void Execute(object parameter)
+		{
+			if (!(parameter is T))
```
Comment (dr1rrb) [inline src/Uno.Foundation/DelegateCommand.T.cs]: If parameter is `null`, this check will fail.

**Pull request merged**